lib/main_common.pm

```diff
@@ -2401,116 +2401,189 @@ sub load_virtualization_tests {
-        loadtest 'virtualization/universal/waitfor_guests';    # Wait for guests to be installed
-
-        loadtest 'virtualization/universal/ssh_guests_init';    # Fetch SSH key from guests and connect
-        loadtest 'virtualization/universal/register_guests';    # Register guests against the SMT server
-        loadtest 'virtualization/universal/upgrade_guests';    # Upgrade all guests
-        loadtest 'virtualization/universal/patch_guests';    # Apply patches to all compatible guests
-        loadtest 'virtualization/universal/patch_and_reboot';    # Apply updates and reboot
-
+    # QAM VM installation
+    if (check_var('ENABLE_VM_INSTALL', 1)) {
         loadtest "virt_autotest/login_console";
-    }
-
-    loadtest "virtualization/universal/list_guests" unless ($windows);    # List all guests and ensure they are running
-
-    if (check_var('VIRT_PART', 'install')) {
-        loadtest "virtualization/universal/kernel";    # Virtualization kernel functions
-    }
-
-    if (check_var('VIRT_PART', 'virtmanager')) {
-        loadtest 'virtualization/universal/virtmanager_init';    # Connect to the Xen hypervisor using virt-manager
-        loadtest 'virtualization/universal/virtmanager_offon';    # Turn all VMs off and then on again
-
-        if (is_sle('12-SP3+')) {
-            loadtest 'virtualization/universal/virtmanager_add_devices';    # Add some aditional HV to all VMs
-            loadtest 'virtualization/universal/virtmanager_rm_devices';    # Remove the aditional HV from all VMs
+        # Prepare and wait for guests
+        loadtest "virtualization/universal/prepare_guests";
+        loadtest "virtualization/universal/waitfor_guests";
+        if (check_var('PATCH_WITH_ZYPPER', 1)) {
+            # Patch system and reboot
+            loadtest "virtualization/universal/patch_and_reboot";
+            if (my $update_package = get_var('UPDATE_PACKAGE')) {
+                if ($update_package eq 'kernel-default') {
+                    # If update package is 'kernel-default', list and patch guests
+                    loadtest "virt_autotest/login_console";
+                    loadtest "virtualization/universal/list_guests";
+                    loadtest "virtualization/universal/patch_guests";
+                } elsif ($update_package eq 'xen' || $update_package eq 'qemu') {
+                    # If update package is 'xen' or 'qemu', list guests
+                    loadtest "virt_autotest/login_console";
+                    loadtest "virtualization/universal/list_guests";
+                }
+            }
+        }
+        # Run kernel tests and finish
+        loadtest "virtualization/universal/kernel";
+        loadtest "virtualization/universal/finish";
+    }
+
+    # Only execute sriov test if ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH is 1
+    if (check_var('ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH', 1) && is_sle('>15')) {
```

Comment:
`get_var()` is used more commonly for these boolean settings of "Enable_xxx".

Reply:
Updated.
From my perspective, if debug, check_var is much better.
using get_var this time.